pkg/controllers/kluctldeployment_controller_status.go

```diff
@@ -35,6 +37,20 @@ func (r *KluctlDeploymentReconciler) patchReadyCondition(ctx context.Context, ob
 
 // patchFail returns the original error + patchErr if required
 func (r *KluctlDeploymentReconciler) patchFail(ctx context.Context, obj *kluctlv1.KluctlDeployment, reason string, err error) error {
+	var lastDeployResult *result.CommandResultSummary
+	if obj.Status.LastDeployResult != nil {
+		parseErr := yaml.ReadYamlBytes(obj.Status.LastDeployResult.Raw, &lastDeployResult)
+		if parseErr != nil {
+			log.Info(fmt.Sprintf("Failed to parse last deploy result: %s", parseErr.Error()))
+		}
+	}
+
+	startTime := time.Now().Unix()
+	if lastDeployResult != nil {
+		startTime = lastDeployResult.Command.StartTime.Unix()
+	}
+
+	internal_metrics.NewKluctlLastDeployStartTime(obj.Namespace, obj.Name).Set(float64(startTime))
```

Comment:
Better do the conversion to int64 here

Reply:
The metric is a gauge which needs a float64 value and timestamps should be gauges as I found

Reply:
Got it now, all time -> int64 -> float64 conversion happens at the metrics creation